app/service/file_svc.py

```diff
@@ -54,10 +54,11 @@ def __init__(self, plugins, exfil_dir):
 
     """ PRIVATE """
 
-    async def _create_unique_exfil_sub_directory(self):
-        dir_name = str(uuid.uuid4())
+    async def _create_exfil_sub_directory(self, headers):
```

Comment:
will need to see an example with this

Reply:
Sure, to run this you can 
add 
$web.setRequestHeader("X-Request-ID","this_request_id");
to ea713bc4-63f0-491c-9a6f-0b01d560b87e (Exfil Staged Directory)
and run an adversary that uses this exfiltration technique.
this will create a directory named "this_request_id" in the /tmp directory where exfil'd files will be dropped.

since the file upload endpoint can be hit outside of an operation it allows the exfil directory to always be tied back to a request

Reply:
let me think on this one. I like what ur shooting for, I'm just thinking through the request-ID header and default uuid dirs.

Reply:
can you update the exfil ability to use this, as part of this PR? this gives us one working example that is cross-platform.

Reply:
https://github.com/mitre/stockpile/pull/126

Reply:
one adjustment on that, then it's good 2 go